### ISSUE TITLE
Investigate performance improvements from implementing pipeline caching for Nuget restore in CI Pipelines

### ISSUE THREAD

**Issue by mrivera-ms** [microsoft/botbuilder-dotnet#4410]
Nuget restore takes nearly 17% of the average pipeline duration. [Azure Pipelines Cache task for Nuget](https://user-images.githubusercontent.com/44449640/89656478-7d6e1000-d880-11ea-8740-e4a5a4037d6a.png) is an option we should try to reduce duration. 

![image](https://user-images.githubusercontent.com/44449640/89656478-7d6e1000-d880-11ea-8740-e4a5a4037d6a.png)

**Comment by denscollo** [comment]
Hi @mrivera-ms ! We investigated the use of the Azure Pipelines Cache task and concluded that it doesn't contribute with a reduction of time for the CI pipeline. 
We ran some tests to get an average time using the Cache Task and without it and we came to the conclusion that it doesn't reduce the total time, as can be seen in the tables below.
![imagen](https://user-images.githubusercontent.com/27219455/90016697-c824c900-dc80-11ea-82c0-cb72e858171f.png)

Also, the cache created with the task is immutable (known issue), which means that it can't be updated when the packages change or must be cleared. One workaround for this is to create a key that changes after certain amount of time, creating a new cache with the updated dependencies.

In conclusion, we don't think using the Azure Pipelines Cache task for the NuGet packages is a good idea, as it doesn't reduce the execution time and adds complexity to manage the caches

**Comment by mrivera-ms** [comment]
Thank you for the analysis @denscollo, I will close this issue as no further action is needed.
